File: leetcode-part-1/java/412_Fizz_Buzz.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> fizzBuzz(int n) {
        List<String> res = new ArrayList<>();
        for (int i = 1; i <= n; i++) {
            String tmp = "";
            if (i % 3 == 0) tmp += "Fizz";
            if (i % 5 == 0) tmp += "Buzz";
            if (tmp.length() == 0) tmp += String.valueOf(i);
            res.add(tmp);
        } 
        return res;
    }
}

File: leetcode-part-1/cpp/412_Fizz_Buzz.cpp
#define pb push_back
class Solution {
public:
    vector<string> fizzBuzz(int n) {
        int i;
        vector<string> s;
        for (i = 0; i < n; i++) {
            if ((i + 1) % 15 == 0)
                s.pb("FizzBuzz");
            else if ((i + 1) % 5 == 0)
                s.pb("Buzz");
            else if ((i + 1) % 3 == 0)
                s.pb("Fizz");
            else
                s.pb(to_string(i + 1));
        }
        return s;
    }
};
